src/main/java/com/thoughtworks/gauge/datastore/ScenarioDataStore.java

```diff
@@ -0,0 +1,41 @@
+/*----------------------------------------------------------------
+ *  Copyright (c) ThoughtWorks, Inc.
+ *  Licensed under the Apache License, Version 2.0
+ *  See LICENSE.txt in the project root for license information.
+ *----------------------------------------------------------------*/
+
+package com.thoughtworks.gauge.datastore;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+public class ScenarioDataStore {
+    private static ThreadLocal<ConcurrentHashMap<Object, Object>> map = ThreadLocal.withInitial(ConcurrentHashMap::new);
+
+    /**
+     * @param key   - Key of the data entry
+     * @param value - value of the Data entry
+     */
+    public static synchronized void put(Object key, Object value) {
+        map.get().put(key, value);
```

Comment:
Now that we've changed from `HashMap` to `ConcurrentHashMap`, it throws a null pointer when setting  null value. It would be good to check for null and skip the put.

Reply:
Message: java.lang.NullPointerException
Stack Trace:
java.base/java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:1011)
java.base/java.util.concurrent.ConcurrentHashMap.put(ConcurrentHashMap.java:1006)
com.thoughtworks.gauge.datastore.ScenarioDataStore.put(ScenarioDataStore.java:19)